tests/framework/pvc_utils.go

```diff
@@ -0,0 +1,192 @@
+package framework
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultPvcMountPath = "/pvc"
+	poll                = 2 * time.Second
+)
+
+// Creates a PVC in the passed in namespace from the passed in PersistentVolumeClaim definition.
+// An example of creating a PVC without annotations looks like this:
+// CreatePVCFromDefinition(client, namespace, NewPVCDefinition(name, size, nil))
+func (f *Framework) CreatePVCFromDefinition(namespace string, def *k8sv1.PersistentVolumeClaim) (*k8sv1.PersistentVolumeClaim, error) {
+	var pvc *k8sv1.PersistentVolumeClaim
+	err := wait.PollImmediate(poll, DefaultTimeout, func() (bool, error) {
+		var err error
+		pvc, err = f.K8sClient.CoreV1().PersistentVolumeClaims(namespace).Create(def)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		return false, err
+	})
+	if err != nil {
+		return nil, err
+	}
+	return pvc, nil
```

Comment:
so we know here there are no api errors but the pvc may not yet be created.

Reply:
So that is an interesting question, with a complex answer. PVC phase has 3 options. Pending/Bound/Lost. And depending on the type of underlying storage Bound may not happen until a pod attaches to the PVC (in particular local volume storage, as the PV used will be on the host the pod is scheduled on). Or it might be bound immediately to the PV (most other storage). So what exactly will we wait on? We can't check the phase, unless we want to check for either Pending or Bound, but I believe the phase goes to Pending immediately so there is nothing to wait on.

Reply:
Phase changes are not necessarily ordered (a->b->c). Sometimes, depending on randomness in the cluster, the Phase might go from a->c completely skipping b.
Nevermind, thinking more I think this code is fine. Eg. Bound may never be seen if Phase is set to Lost right after being set to Bound. This is fine and the poll with expire. It's also possible that the poll sees Bound but right after that Phase changes to Lost...